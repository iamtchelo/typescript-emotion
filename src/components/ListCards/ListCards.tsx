import * as React from 'react';
import { ListCardsWrapper } from './list-cards.elements';

export interface ListCardsProps {
  renderType?: 'grid' | 'list';
}

const ListCards: React.SFC<ListCardsProps> = ({ renderType = 'grid', children }) => (
  <ListCardsWrapper renderType={renderType}>
    {children}
  </ListCardsWrapper>
);

export default ListCards;
