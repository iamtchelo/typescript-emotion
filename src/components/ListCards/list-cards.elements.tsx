import styled, { css } from 'react-emotion';
import { ListCardsProps } from './ListCards';

export const ListCardsWrapper = styled('div')`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;

  ${({ renderType }: ListCardsProps) => renderType === 'grid' && css`
    flex-direction: row
  `};

  ${({ renderType }: ListCardsProps) => renderType === 'list' && css`
    align-content: center;
    flex-direction: column;

    & > div {
      width: 350px;
      height: 300px;
      border: 5px solid white;

      &:nth-child(even):hover,
      &:nth-child(odd):hover {
        transform: none;
      }

      & > img {
        width: 120px;
        border: none;
      }

      & > h4 {
        padding: 15px 0;
      }
    }
  `};
`;
