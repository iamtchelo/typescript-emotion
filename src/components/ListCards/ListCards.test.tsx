import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import ListCards from './ListCards';

describe('<ListCards />', () => {
  it('should render grid by default', () => {
    const tree = shallow(<ListCards />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render list', () => {
    const tree = shallow(<ListCards renderType="list" />);
    
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});