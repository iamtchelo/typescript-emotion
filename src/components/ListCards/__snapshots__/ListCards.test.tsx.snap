// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListCards /> should render grid by default 1`] = `
<Styled(div)
  renderType="grid"
/>
`;

exports[`<ListCards /> should render list 1`] = `
<Styled(div)
  renderType="list"
/>
`;
